packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -933,9 +934,50 @@ export class GarbageCollector implements IGarbageCollector {
 	 * cases where objects are used after they are deleted and throw / log errors accordingly.
 	 */
 	public isNodeDeleted(nodePath: string): boolean {
+		//* TODO: Replace usages with getInternalReferenceInfo?
 		return this.deletedNodes.has(nodePath);
 	}
 
+	/**
+	 * Get this client's estimation of the reference state of the give node
+	 * It may be inaccurate in some cases, e.g. if an interactive client sees an object as unreferenced on load,
+	 * some time later it will transition to Inactive even if it is revived in the meantime.
+	 * @param nodePath - Container-relative path to the node
+	 */
+	public getInternalReferenceInfo(nodePath: string): IFluidInternalReferenceInfo {
+		//* Is this a fair requirement?
+		//* PS - It's assert not UsageError because this class isn't exported, so callers are internal and should know better
+		assert(
+			!this.deletedNodes.has("nodePath"),
+			"Getting internal reference info for a deleted node is not supported",
+		);
+
+		const trackedState = this.unreferencedNodesState.get(nodePath);
+		if (trackedState === undefined) {
+			return { state: "Referenced" };
+		}
+
+		//* Finalize state values and logic here
+		let state: IFluidInternalReferenceInfo["state"];
+		switch (trackedState.state) {
+			case "Active":
+				state = "Referenced";
+				break;
+			case "Inactive":
+			case "SweepReady":
```

Comment:
I wonder if we should return "SweepReady" here. If tombstone feature is not enabled and a node is sweep ready, does it benefit caller to know that this would have been deleted if sweep was enabled? I guess so.